test/framework/src/main/java/org/elasticsearch/test/IndexSettingsModule.java

```diff
@@ -66,6 +66,16 @@ public static IndexSettings newIndexSettings(Index index, Settings settings, Set
         return new IndexSettings(metaData, Settings.EMPTY, (idx) -> Regex.simpleMatch(idx, metaData.getIndex().getName()), new IndexScopedSettings(Settings.EMPTY, settingSet));
     }
 
+    public static IndexSettings newIndexSettings(Index index, Settings settings, IndexScopedSettings indexScopedSettings) {
```

Comment:
I added this method, because I inject a guice managed `IndexScopedSettings` instance in `AbstractQueryTestCase#init()`